### PR TITLE
Correct typos

### PR DESCRIPTION
**Pull request by shlomif**
These commits correct some typos - please consider merging them.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,97 +6,97 @@
 rules including regular expressions, statistical analysis and custom services
 such as URL black lists. Each message is analysed by Rspamd and given a `spam score`.
 
-According to this spam score and the user's settings Rspamd recommends an action for
+According to this spam score and the user's settings, Rspamd recommends an action for
 the MTA to apply to the message, for example, to pass, reject or add a header.
-Rspamd is designed to process hundreds of messages per second simultaneously and has a number of
-features available.
+Rspamd is designed to process hundreds of messages per second simultaneously, and sports a number of
+usefulrfeatures.
```
Comment (moisseev) [inline README.md]: usefulrfeatures →  useful features

File: README.md [hunk 2]
```diff
@@ -6,97 +6,97 @@
...
+words, but instead is organized in chains that are further estimated by the Bayesian classifier. This approach allows to achieve better results than
+traditionally used monograms (or words literally speaking), that are described in details in the following [paper](http://osbf-lua.luaforge.net/papers/osbf-eddc.pdf).
 
 
-* **Fuzzy hashes** - for checking of malicious mail patterns Rspamd uses so called `fuzzy hashes`. Unlike normal hashes, these structures are targeted to hide
-small differences between text patterns allowing to find similar messages quickly. Rspamd has internal storage of such hashes and allows to block mass spam sendings
-quickly based on user's feedback that specifies messages reputation. Moreover, this allows to feed Rspamd with data from [`honeypots`](http://en.wikipedia.org/wiki/Honeypot_(computing)#Spam_versions)
+* **Fuzzy hashes** - for checking of malicious mail patterns, Rspamd uses the so called "fuzzy hashes". Unlike normal hashes, these structures are targeted to hide
+the small differences between text patterns, allowing it to find eimilar messages quickly. Rspamd has an internal storage of such hashes and can block mass spam sendings
```
Comment (moisseev) [inline README.md]: eimilar → similar

File: README.md [hunk 3]
```diff
@@ -4,99 +4,99 @@
...
-such as URL black lists. Each message is analysed by Rspamd and given a `spam score`. 
+such as URL black lists. Each message is analysed by Rspamd and given a `spam score`.
 
-According to this spam score and the user's settings Rspamd recommends an action for
+According to this spam score and the user's settings, Rspamd recommends an action for
 the MTA to apply to the message, for example, to pass, reject or add a header.
-Rspamd is designed to process hundreds of messages per second simultaneously and has a number of 
-features available.
+Rspamd is designed to process hundreds of messages per second simultaneously, and sports a number of
+usefulrfeatures.
```
Comment (fatalbanana) [inline README.md]: That is mangled.

File: README.md [hunk 4]
```diff
@@ -4,99 +4,99 @@
...
 
 
-* **DKIM module** validates a message cryptographic signature against a public key placed in the DNS record of sender's domain. Like SPF,
+* **DKIM module** validates a message’s cryptographic signature against a public key placed in the DNS record of sender's domain. Like SPF,
 this technique is widely spread and allows to validate that a message is sent from that specific domain.
 
 
 * **DNS black lists** allows to estimate reputation of sender's IP address or network. Rspamd uses a number of DNS lists including such lists as
-`SORBS` or `Spamhaus`. However, Rspamd doesn't trust any specific DNS list and use a conjunction of estimations instead that allows to
+`SORBS` or `Spamhaus`. However, Rspamd doesn't trust any specific DNS list, and uses a conjunction of estimations instead that allows to
```
Comment (fatalbanana) [inline README.md]: Use of comma here seems dubious.

File: README.md [hunk 5]
```diff
@@ -4,99 +4,99 @@
...
+Besides its C core, Rspamd provides an extensive [Lua](http://lua.org) API to access almost all the features available directly from C. Lua is an extremely easy
 to learn programming language though it is powerful enough to implement complex mail filters. In fact Rspamd has a significant amount of code written completely in Lua such as
 DNS blacklists checks, user's settings or different maps implementation. You can also write your own filters and rules in Lua adopting Rspamd functionality to your needs.
-Furthermore, Lua programs are very fast and their performance is rather [close](http://attractivechaos.github.io/plb/) to pure `C`. However, you should mention that for the most
-of performance critical tasks you usually use the Rspamd core functionality than Lua code. Anyway, you can also use `LuaJIT` with Rspamd if your goal is maximum performance.
+Furthermore, Lua programs are very fast and their performance is rather [close](http://attractivechaos.github.io/plb/) to pure C. However, you should mention that for the most
```
Comment (fatalbanana) [inline README.md]: Why don't you want `C` & `LuaJIT` to have backticks?

**Comment by shlomif** [comment]
On Thu, 26 Oct 2017 13:30:15 +0000 (UTC)
Alexander Moisseev <notifications@github.com> wrote:

> moisseev requested changes on this pull request.
> 
> 
> 
> >  the MTA to apply to the message, for example, to pass, reject or add a
> > header.  
> -Rspamd is designed to process hundreds of messages per second simultaneously
> and has a number of -features available.
> +Rspamd is designed to process hundreds of messages per second
> simultaneously, and sports a number of +usefulrfeatures.
> 
> usefulrfeatures →  useful features
> 
> >    
>  
> -* **Fuzzy hashes** - for checking of malicious mail patterns Rspamd uses so
> called `fuzzy hashes`. Unlike normal hashes, these structures are targeted to
> hide -small differences between text patterns allowing to find similar
> messages quickly. Rspamd has internal storage of such hashes and allows to
> block mass spam sendings -quickly based on user's feedback that specifies
> messages reputation. Moreover, this allows to feed Rspamd with data from
> [`honeypots`](http://en.wikipedia.org/wiki/Honeypot_(computing)#Spam_versions)
> +* **Fuzzy hashes** - for checking of malicious mail patterns, Rspamd uses
> the so called "fuzzy hashes". Unlike normal hashes, these structures are
> targeted to hide +the small differences between text patterns, allowing it to
> find eimilar messages quickly. Rspamd has an internal storage of such hashes
> and can block mass spam sendings
> 
> eimilar → similar
> 

Thanks! I'll correct it.

-- 
-----------------------------------------------------------------
Shlomi Fish       http://www.shlomifish.org/
Let’s talk about restores instead of backups - http://is.gd/WatQqu

“If it’s not bloat, it’s not us.”, said Richard Stallman, the colourful head of
the GNU project, and started to sing the Free Software song.
    — “The GNU Project Will Integrate GNU Guile into GNU coreutils”

Please reply to list if it's a mailing list post - http://shlom.in/reply .

**Comment by shlomif** [comment]
On Thu, 26 Oct 2017 13:45:27 +0000 (UTC)
Andrew Lewis <notifications@github.com> wrote:

> fatalbanana commented on this pull request.
> 
> 
> 
> >  the MTA to apply to the message, for example, to pass, reject or add a
> > header.  
> -Rspamd is designed to process hundreds of messages per second simultaneously
> and has a number of -features available.
> +Rspamd is designed to process hundreds of messages per second
> simultaneously, and sports a number of +usefulrfeatures.
> 
> That is mangled.
> 

Thanks, will be corrected.

-- 
-----------------------------------------------------------------
Shlomi Fish       http://www.shlomifish.org/
http://www.shlomifish.org/humour/bits/New-versions-of-the-GPL/

Attention: There will be a rain dance Friday night, weather permitting.
    — Source unknown, via Nadav Har’El.

Please reply to list if it's a mailing list post - http://shlom.in/reply .

**Comment by shlomif** [comment]
On Thu, 26 Oct 2017 13:52:00 +0000 (UTC)
Andrew Lewis <notifications@github.com> wrote:

> fatalbanana commented on this pull request.
> 
> 
> 
> >  to learn programming language though it is powerful enough to implement
> > complex mail filters. In fact Rspamd has a significant amount of code
> > written completely in Lua such as  
>  DNS blacklists checks, user's settings or different maps implementation. You
> can also write your own filters and rules in Lua adopting Rspamd
> functionality to your needs. -Furthermore, Lua programs are very fast and
> their performance is rather [close](http://attractivechaos.github.io/plb/) to
> pure `C`. However, you should mention that for the most -of performance
> critical tasks you usually use the Rspamd core functionality than Lua code.
> Anyway, you can also use `LuaJIT` with Rspamd if your goal is maximum
> performance. +Furthermore, Lua programs are very fast and their performance
> is rather [close](http://attractivechaos.github.io/plb/) to pure C. However,
> you should mention that for the most
> 
> Why don't you want `C` & `LuaJIT` to have backticks?
> 

Because in markdown syntax this becomes the
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/code element. If you
want, you can use UTF-8 quotes (“ ”  ‘) but there is no need for them for C and
LuaJIT - see https://en.wikipedia.org/wiki/C_%28programming_language%29 .


-- 
-----------------------------------------------------------------
Shlomi Fish       http://www.shlomifish.org/
http://shlomifishswiki.branchable.com/Self-Sufficiency/

Chuck Norris can end world hunger, but he thinks that hungry people make
humanity a more challenging adversary.
    — http://www.shlomifish.org/humour/bits/facts/Chuck-Norris/

Please reply to list if it's a mailing list post - http://shlom.in/reply .

**Comment by shlomif** [comment]
Hi! Any news? Can this be merged already?

**Comment by shlomif** [comment]
Ping! Can this be merged now?

File: README.md [hunk 6]
```diff
@@ -4,99 +4,99 @@
...
-such as URL black lists. Each message is analysed by Rspamd and given a `spam score`. 
+such as URL black lists. Each message is analysed by Rspamd and given a `spam score`.
 
-According to this spam score and the user's settings Rspamd recommends an action for
+According to this spam score and the user's settings, Rspamd recommends an action for
 the MTA to apply to the message, for example, to pass, reject or add a header.
-Rspamd is designed to process hundreds of messages per second simultaneously and has a number of 
-features available.
+Rspamd is designed to process hundreds of messages per second simultaneously, and sports a number of
+useful features.
```
Comment (vstakhov) [inline README.md]: What's the purpose of this change? I have never seen the word 'sports' used in this context. Another change is also meaningless to my opinion.

**Comment by vstakhov** [comment]
I'm sorry for a long response - I've just added couple of comments as well.

**Comment by shlomif** [comment]
On Sat, 16 Dec 2017 10:19:35 -0800
Vsevolod Stakhov <notifications@github.com> wrote:

> vstakhov commented on this pull request.
> 
> 
> 
> >  the MTA to apply to the message, for example, to pass, reject or add a
> > header.  
> -Rspamd is designed to process hundreds of messages per second simultaneously
> and has a number of -features available.
> +Rspamd is designed to process hundreds of messages per second
> simultaneously, and sports a number of +useful features.
>  
> 
> What's the purpose of this change? I have never seen the word 'sports' used
> in this context. Another change is also meaningless to my opinion.
> 

the previous version sounds like a non-idiomatic "dead-end"/pointless sentence.
Regarding "sports" see the third meaning at
https://en.wiktionary.org/wiki/sport#Verb .

-- 
-----------------------------------------------------------------
Shlomi Fish       http://www.shlomifish.org/
Parody of "The Fountainhead" - http://shlom.in/towtf

  <rindolf>  Barbapappa: Buffy Summers > Chuck Norris
<preaction>  so, Chuck Norris is a vampire?
  <rindolf>  preaction: no, but in the Buffyverse, he’s getting pwned by Buffy
    — #perl, http://is.gd/grdEqT

Please reply to list if it's a mailing list post - http://shlom.in/reply .

**Comment by vstakhov** [comment]
Well, I've asked a native English speaker about usage of `sport` verb here and he told me that it doesn't sound familiar to him as well. I'd suggest to leave here `uses`, `utilises` or even `provides` verbs...

**Comment by shlomif** [comment]
@vstakhov : see my new commit.

**Comment by vstakhov** [comment]
Cool, thank you! Merging.

**Comment by shlomif** [comment]
On Sun, 17 Dec 2017 09:54:40 +0000 (UTC)
Vsevolod Stakhov <notifications@github.com> wrote:

> Merged #1884.
> 

thanks, and you are welcome.

-- 
-----------------------------------------------------------------
Shlomi Fish       http://www.shlomifish.org/
Optimising Code for Speed - http://shlom.in/optimise

Friends may come and go, but enemies accumulate.
    — Thomas Jones (1892-1969)

Please reply to list if it's a mailing list post - http://shlom.in/reply .
